shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java

```diff
@@ -176,24 +181,34 @@ private void processingComplete(Class<?> entity) {
      * @throws IllegalStateException in case of an invalid HTTP response
      */
     private void validateHttpResponseBeforeParsing(HttpResponse<Buffer> httpResponse) {
+        if (httpResponse.statusCode() == 401) {
```

Comment:
do we really need this? is it not the check when we create the request enough?

Reply:
Nope, the token can expire in the meanwhile

Reply:
yes, but which is the effect? I understood that if the token expired just in that small interval (between the check to validate the token is active and the actual call) then we fire a token refreshal here, but the failed call, is it performed again?
If not performed again, it will return a 401 to the caller and the next call will try to refresh the token anyway (because the isTokenExpired will return true in that case), so thats Im wodering is this is really needed.

Reply:
I'm implementing the retry mechanism in another pull request

Reply:
I'll rebase this one once I'm ready with the other one